### ISSUE TITLE
Issue with Song Title not Changing when Switching Song

### ISSUE THREAD

**Issue by jblais493** [mihirlad55/polybar-spotify-module#20]
I am sure there is a simple fix - my issue is that upon switching song, the text doesn't alter; in order for the text to be updates I have to restart the spotify-listener serveice, but then run into the same issue upon song switch.

Any help would be great appreciated, great module!

**Comment by mihirlad55** [comment]
The song title should be changing automatically as long as `spotify-listener.service` is running in the background. Are you using spotify or spotifyd?

Could you please post your polybar config?
Please wrap your config output in:
```
<details>
  <summary>Click to expand</summary>
  insert polybar config here
</details>
```

**Comment by jblais493** [comment]
<details>
  <summary>Click to expand</summary>
  ;==========================================================
;
;
;   ██████╗  ██████╗ ██╗  ██╗   ██╗██████╗  █████╗ ██████╗
;   ██╔══██╗██╔═══██╗██║  ╚██╗ ██╔╝██╔══██╗██╔══██╗██╔══██╗
;   ██████╔╝██║   ██║██║   ╚████╔╝ ██████╔╝███████║██████╔╝
;   ██╔═══╝ ██║   ██║██║    ╚██╔╝  ██╔══██╗██╔══██║██╔══██╗
;   ██║     ╚██████╔╝███████╗██║   ██████╔╝██║  ██║██║  ██║
;   ╚═╝      ╚═════╝ ╚══════╝╚═╝   ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═╝
;
;
;   To learn more about how to configure Polybar
;   go to https://github.com/polybar/polybar
;
;   The README contains a lot of information
;
;==========================================================

[colors]
;background = ${xrdb:color0:#222}
background = #0000000 
background-alt = #444
;foreground = ${xrdb:color7:#222}
foreground = #ffffff 
foreground-alt = #555
primary = #ffffff
secondary = #ffffff
alert = #bd2c40

[bar/example]
monitor = ${env:MONITOR:DP-4}
width = 100%
height = 35
;offset-x = 1%
;offset-y = 1%
radius = 0
fixed-center = true
enable-ipc = true

background = #00000000
foreground = #ffffff 

line-size = 3
line-color = #f00

border-size = 0
border-color = #00000000

padding-left = 0
padding-right = 2

module-margin-left = 1
module-margin-right = 1 

;font-0 = "NotoSansMono Light Nerd Font Complete":pixelsize=16;1
;font-0 = Saucecode Pro Nerd Font:style=semibold;pixelsize=20;1
;font-0 = Montserrat:style=Light:size=16;1
font-0 = JetBrainsMono Nerd Font:style=semibold:pixelsize=18;3
font-1 = Weather Icons:style=Regular:size=18;1
font-2 = Sarasa Gothic HC:style=semibold:pixelsize=18;1
font-3 = Font Awesome 5 Free:style=Regular:pixelsize=16;1
font-4 = Font Awesome 5 Free:style=Solid:pixelsize=16;1
font-5 = Font Awesome 5 Brands:style=Regular:pixelsize=16;1
font-6 = Material Icons:style=Bold:pixelsize=18;1

modules-left = bspwm
modules-center = mpd spotify previous playpause next
modules-right = weather temperature pulseaudio battery wlan bluetooth date powermenu

tray-position = right
tray-padding = 2
;tray-background = #0063ff

wm-restack = bspwm
;wm-restack = i3

;override-redirect = true

;scroll-up = bspwm-desknext
;scroll-down = bspwm-deskprev

;scroll-up = i3wm-wsnext
;scroll-down = i3wm-wsprev

cursor-click = pointer
cursor-scroll = ns-resize

[module/xwindow]
type = internal/xwindow
label = %title:0:30:...%

[module/bspwm]
type = internal/bspwm

label-font = 2
pin-workspaces = true
inline-mode = false
enable-click = true
enable-scroll = true
reverse-scroll = true
fuzzy-match = true

format = <label-state> <label-mode>

label-dimmed-foreground = #ffffff

label-dimmed-occupied = %icon%
label-dimmed-occupied-foreground = #626483
label-dimmed-occupied-padding = 2

label-dimmed-empty = %icon%
label-dimmed-empty-foreground = #3a3c4e
label-dimmed-empty-padding = 2

label-focused = %icon%
label-focused-foreground = #f8f8f2
label-focused-background = #ea3a3c4e
label-focused-underline = #ffffff
label-focused-padding = 2

label-occupied = %icon%
label-occupied-foreground = #ffffff
label-occupied-padding = 2 

label-urgent = %icon%
label-urgent-foreground = #f8f8f2
label-urgent-background = #eaff5555
label-urgent-padding = 2 

label-empty = %icon%
label-empty-foreground = #ffffff
label-empty-padding = 2 

label-monocle = 
label-monocle-foreground = #ffffff
label-tiled = 
label-tiled-foreground = #ffffff
label-fullscreen = 
label-fullscreen-foreground = #ffffff
label-floating = 
label-floating-foreground = #ffffff 
label-pseudotiled = P
label-pseudotiled-foreground = #ffffff
label-locked = " "
label-locked-foreground = #ffffff
label-sticky = " "
label-sticky-foreground = #ffffff
label-private = " "
label-private-foreground = #ffffff
ws-icon-0 = 1;一
ws-icon-1 = 2;二
ws-icon-2 = 3;三
ws-icon-3 = 4;四
ws-icon-4 = 5;五
ws-icon-5 = 6;六
ws-icon-6 = 7;七
ws-icon-7 = 8;八
ws-icon-8 = 9;九
;                 

[module/bluetooth]
type = custom/script
exec = ~/.config/polybar/scripts/bluetooth.sh
interval = 2
click-left = exec blueberry
click-right = exec ~/.config/polybar/scripts/toggle_bluetooth.sh
format-padding = 0 
format-background = #00000000
format-foreground = #ffffff


[module/weather]
type = custom/script
exec = ~/.config/polybar/polybar-forecast-linux-x86_64 
exec-if = ping openweathermap.org -c 1
interval = 600
label-font = 2 

[module/mpd]
type = internal/mpd
label-song-foreground = ${colors.primary}
format-online =<icon-stop> <label-song>

icon-prev = 
icon-stop = 
icon-play = 
icon-pause = 
icon-next = 

label-song-maxlen = 50 
label-song-ellipsis = true

;[module/mpd]
;type = internal/mpd
;format-online = <label-song>  <icon-prev> <icon-stop> <toggle> <icon-next>
;
;icon-prev = 
;icon-stop = 
;icon-play = 
;icon-pause = 
;icon-next = 
;
;label-song-maxlen = 25
;label-song-ellipsis = true

[module/xbacklight]
type = internal/xbacklight

format = <label> <bar>
label = BL

bar-width = 10
bar-indicator = |
bar-indicator-foreground = #fff
bar-indicator-font = 2
bar-fill = ─
bar-fill-font = 2
bar-fill-foreground = #9f78e1
bar-empty = ─
bar-empty-font = 2
bar-empty-foreground = ${colors.foreground-alt}

[module/backlight-acpi]
inherit = module/xbacklight
type = internal/backlight
card = intel_backlight


[module/cpu]
type = internal/cpu
interval = 2
format-prefix = "  "
format-prefix-foreground = ${colors.secondary}
format = <label> <ramp-load>
label = %percentage:2%%

; Spacing between individual per-core ramps
ramp-load-0 = ▁
ramp-load-1 = ▂
ramp-load-2 = ▃
ramp-load-3 = ▄
ramp-load-4 = ▅
ramp-load-5 = ▆
ramp-load-6 = ▇
ramp-load-7 = █

[module/memory]
type = internal/memory
interval = 2
format-prefix = " "
format-prefix-foreground = ${colors.secondary}
format = <label> <ramp-used>
label = %gb_used%

; Only applies if <ramp-used> is used
ramp-used-0 = ▁
ramp-used-1 = ▂
ramp-used-2 = ▃
ramp-used-3 = ▄
ramp-used-4 = ▅
ramp-used-5 = ▆
ramp-used-6 = ▇
ramp-used-7 = █

ramp-used-foreground = ${colors.secondary}


[module/wlan]
type = internal/network
interface = wlp4s0
interval = 3.0

format-connected = <ramp-signal> <label-connected>

; Default: %ifname% %local_ip%
label-font = 6 
label-connected = %essid% 
label-connected-foreground = #ffffff

; Default: (none)
label-disconnected = not connected
label-disconnected-foreground = #66ffffff

; Default: (none)
;label-packetloss = %essid%
;label-packetloss-foreground = #eefafafa

; Only applies if <ramp-signal> is used
; 
ramp-signal-0 = 
ramp-signal-1 = 
ramp-signal-2 = 
ramp-signal-3 = 
ramp-signal-4 = 
ramp-signal-5 = 
ramp-signal-6 = 
ramp-signal-7 = 



ramp-signal-foreground = ${colors.primary}

[module/eth]
type = internal/network
interface = enp0s25
interval = 3.0

format-connected-underline = #55aa55
format-connected-prefix = " "
format-connected-prefix-foreground = ${colors.foreground-alt}
label-connected = %local_ip%

format-disconnected =
;format-disconnected = <label-disconnected>
;format-disconnected-underline = ${self.format-connected-underline}
;label-disconnected = %ifname% disconnected
;label-disconnected-foreground = ${colors.foreground-alt}

[module/date]
type = internal/date
interval = 1 

date =
;date-alt = " %Y-%m-%d"
date-alt = " %m-%d"

time = %H:%M
time-alt = %H:%M

format-prefix = 
format-prefix-foreground = ${colors.foreground-alt}
label-font = 1
label = %date% %time%

[module/pulseaudio]
type = internal/pulseaudio

; Sink to be used, if it exists (find using `pacmd list-sinks`, name field)
; If not, uses default sink
;sink = alsa_output.pci-0000_12_00.3.analog-stereo

; Use PA_VOLUME_UI_MAX (~153%) if true, or PA_VOLUME_NORM (100%) if false
; Default: true
use-ui-max = true

format-volume = <ramp-volume> <label-volume>

label-font = 6 
label-volume = %percentage%%
label-volume-foreground = ${colors.foreground}

label-muted = muted
label-muted-foreground = ${colors.foreground}

; Only applies if <ramp-volume> is used
ramp-volume-0 = 
ramp-volume-1 = 
ramp-volume-2 = 

ramp-volume-foreground = #ffffff 



;[module/pulseaudio]
;type = internal/pulseaudio
;
;format-volume = <label-volume> <bar-volume>
;label-volume = VOL %percentage%%
;label-volume-foreground = ${root.foreground}
;
;label-muted = 🔇 muted
;label-muted-foreground = #666
;
;bar-volume-width = 10
;bar-volume-foreground-0 = #55aa55
;bar-volume-foreground-1 = #55aa55
;bar-volume-foreground-2 = #55aa55
;bar-volume-foreground-3 = #55aa55
;bar-volume-foreground-4 = #55aa55
;bar-volume-foreground-5 = #f5a70a
;bar-volume-foreground-6 = #ff5555
;bar-volume-gradient = false
;bar-volume-indicator = |
;bar-volume-indicator-font = 2
;bar-volume-fill = ─
;bar-volume-fill-font = 2
;bar-volume-empty = ─
;bar-volume-empty-font = 2
;bar-volume-empty-foreground = ${colors.foreground-alt}

[module/alsa]
type = internal/alsa

format-volume = <label-volume> <bar-volume>
label-volume = VOL
label-volume-foreground = ${root.foreground}

format-muted-prefix = " "
format-muted-foreground = ${colors.foreground-alt}
label-muted = sound muted

bar-volume-width = 10
bar-volume-foreground-0 = #55aa55
bar-volume-foreground-1 = #55aa55
bar-volume-foreground-2 = #55aa55
bar-volume-foreground-3 = #55aa55
bar-volume-foreground-4 = #55aa55
bar-volume-foreground-5 = #f5a70a
bar-volume-foreground-6 = #ff5555
bar-volume-gradient = false
bar-volume-indicator = |
bar-volume-indicator-font = 2
bar-volume-fill = ─
bar-volume-fill-font = 2
bar-volume-empty = ─
bar-volume-empty-font = 2
bar-volume-empty-foreground = ${colors.foreground-alt}



[module/battery]
type = internal/battery

; This is useful in case the battery never reports 100% charge
full-at = 95

; Use the following command to list batteries and adapters:
; $ ls -1 /sys/class/power_supply/
battery = BAT0
adapter = ADP1

; If an inotify event haven't been reported in this many
; seconds, manually poll for new values.
;
; Needed as a fallback for systems that don't report events
; on sysfs/procfs.
;
; Disable polling by setting the interval to 0.
;
; Default: 5
;poll-interval = 5
; see "man date" for details on how to format the time string
; NOTE: if you want to use syntax tags here you need to use %%{...}
; Default: %H:%M:%S
;label-prefix =  

; Available tags:
;   <label-charging> (default)
;   <bar-capacity>
;   <ramp-capacity>
;   <animation-charging>
format-charging = <animation-charging> <label-charging>

; Available tags:
;   <label-discharging> (default)
;   <bar-capacity>
;   <ramp-capacity>
;   <animation-discharging>
format-discharging = <ramp-capacity> <label-discharging>
; label-discharging = %percentage%

; Available tags:
;   <label-full> (default)
;   <bar-capacity>
;   <ramp-capacity>
format-full = <ramp-capacity> <label-full>

; Available tokens:
;   %percentage% (default) - is set to 100 if full-at is reached
;   %percentage_raw%
;   %time%
;   %consumption% (shows current charge rate in watts)
label-charging = %percentage%%

; Available tokens:
;   %percentage% (default) - is set to 100 if full-at is reached
;   %percentage_raw%
;   %time%
;   %consumption% (shows current discharge rate in watts)
label-discharging = %percentage%%

; Available tokens:
;   %percentage% (default) - is set to 100 if full-at is reached
;   %percentage_raw%
label-full = Fully charged

; Only applies if <ramp-capacity> is used
;
ramp-capacity-0 = 
ramp-capacity-1 = 
ramp-capacity-2 = 
ramp-capacity-3 = 
ramp-capacity-4 = 

; Only applies if <bar-capacity> is used
;bar-capacity-width = 10

; Only applies if <animation-charging> is used
animation-charging-0 = 
animation-charging-1 = 
animation-charging-2 = 
animation-charging-3 = 
animation-charging-4 = 
; Framerate in milliseconds
animation-charging-framerate = 750

; Only applies if <animation-discharging> is used
animation-discharging-0 = 
animation-discharging-1 = 
animation-discharging-2 = 
animation-discharging-3 = 
animation-discharging-4 = 
; Framerate in milliseconds
animation-discharging-framerate = 500


;

[module/temperature]
type = internal/temperature
thermal-zone = 1
warn-temperature = 60

;<ramp>
format = <label>
format-prefix = " "
;format-underline = #f50a4d
format-warn = <ramp> <label-warn>
;format-warn-underline = ${self.format-underline}
format-warn-underline = #f50a4d

label = %temperature-c%
label-warn = %temperature-c%
label-warn-foreground = ${colors.secondary}

ramp-0 = 
ramp-1 = 
ramp-2 = 
ramp-foreground = ${colors.foreground-alt}

[module/powermenu]
type = custom/menu

expand-right = true

format-spacing = 1

label-open = 
label-open-foreground = #fffff
label-close =  cancel
label-close-foreground = #e06c75
label-separator = |
label-separator-foreground = ${colors.foreground}

menu-0-0 = reboot
menu-0-0-exec = menu-open-1
menu-0-1 = power off
menu-0-1-exec = menu-open-2

menu-1-0 = cancel
menu-1-0-exec = menu-open-0
menu-1-1 = reboot
menu-1-1-exec = sudo reboot

menu-2-0 = power off
menu-2-0-exec = sudo poweroff
menu-2-1 = cancel
menu-2-1-exec = menu-open-0

;[module/powermenu]
;type = custom/menu
;
;expand-right = true
;
;format-spacing = 1
;
;label-open = 
;label-open-foreground = ${colors.secondary}
;label-close =  cancel
;label-close-foreground = ${colors.secondary}
;label-separator = |
;label-separator-foreground = ${colors.foreground-alt}
;
;menu-0-0 = reboot
;menu-0-0-exec = menu-open-1
;menu-0-1 = power off
;menu-0-1-exec = menu-open-2
;
;menu-1-0 = cancel
;menu-1-0-exec = menu-open-0
;menu-1-1 = reboot
;menu-1-1-exec = sudo reboot
;
;menu-2-0 = power off
;menu-2-0-exec = sudo poweroff
;menu-2-1 = cancel
;menu-2-1-exec = menu-open-0

[settings]
screenchange-reload = true
;compositing-background = xor
;compositing-background = screen
;compositing-foreground = source
;compositing-border = over
;pseudo-transparency = false

[global/wm]
margin-top = 0 
margin-bottom = 0 

; vim:ft=dosini

[module/previous]
format-font = 5
type = custom/ipc
; Default
hook-0 = echo ""
; When spotify active
hook-1 = echo ""

click-left = "spotifyctl -q previous"


[module/next]
format-font = 5
type = custom/ipc
; Default
hook-0 = echo ""
; When spotify active
hook-1 = echo ""
click-left = "spotifyctl -q next"


[module/playpause]
format-font = 5
type = custom/ipc
; Default
hook-0 = echo ""
; Playing
hook-1 = echo ""
; Paused
hook-2 = echo ""

click-left = "spotifyctl -q playpause"


[module/spotify]
format-font = 1
type = custom/ipc
; Default
hook-0 = echo ""
; Playing/paused show song name and artist
hook-1 = spotifyctl -q status --format '%artist%: %title%'

;icon-prev = 
;icon-stop = ""
;icon-play = 
;icon-pause = 
;icon-next = 
</details>

As an addendum, the module change occasionally if I let the song play to the end, but it's random if it does or doesn't. Weirdly enough this config works perfectly on my laptop with songs following perfectly. May just be a bug with the font I am using?
